drafts/25-revise-24-177 Fortran preprocessor requirements.txt

```diff
@@ -58,13 +58,22 @@ from C and Fortran syntax in the following ways:
     7. FPP recognizes /* ... */ C-style comments on directive lines.
     8. Expressions in #if and #elif directives contain constructs from
        both Fortran and CPP.
-    9. Expressions in #if and #elif directives that evaluate to LOGICAL
-       or INTEGER values. .TRUE. or a non-zero integer are considered
-       true.
-    10. Undefined identifiers are treated as zero, as in CPP.
-    11. Integer expressions in preprocessor directives are calculated in
-       the maximum precision the processor supports. (There are no KIND
-       specifiers on integer constants in the preprocessor.)
+    9. Expressions in #if and #elif directives evaluate to INTEGER values.
+       Zero values are treated as 'false'. Non-zero values are treated as
+       'true'.
```

Comment:
When it comes time to actually write spec wording for this, it's not enough to say the expression "evaluate to INTEGER values". There must be additional constraints to rule out (for example) character strings in preprocessor conditional expressions.

For now it might be sufficient to say something like 
```suggestion
    9. Expressions in #if and #elif directives must be integer constant expressions as in CPP
       (with the extensions described below), and evaluate to INTEGER values.
       As in CPP, zero values are treated as 'false'. Non-zero values are treated as
       'true'.
```

Reply:
Will incorporate this change.